backend/matching-service/app/src/matching/matching.interfaces.ts

```diff
@@ -0,0 +1,17 @@
+import { Socket } from "socket.io";
+
+export interface Preferences {
+  difficulty: string[];
```

Comment:
we could make this an enum (e.g. `DifficultyType` can look in the frontend for the implementation)

but do u think that is necessary? cos technically the matching service doesn't have to know the details (preferences could even be an `any` object that just does a 1:1 comparison). it only has to know about the implementation if we want to do more complex matching logic (e.g. match if have max of 1 different field)

Reply:
modified the comparison to perform a "subset or equal to" check for arrays in 65b4421e074f02d4424c552d15138885f99ac931